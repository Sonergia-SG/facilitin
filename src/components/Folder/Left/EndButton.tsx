import React from 'react';
import { connect } from 'react-redux';

import { folderEnding } from '../../../store/actions/views/folder';

import { OperationFull } from '../../../store/reducer/entities/types';
import { FolderPendingItem } from '../../../store/reducer/views/folder/types';

interface Props {
  data: OperationFull;
  pending: FolderPendingItem | undefined;
  ending: any;
  locked: boolean;
}

export const EndButtonComponent = ({
  data, ending, pending, locked,
}: Props) => {
  const displayButton = data.dossierprimefile
    ? data.dossierprimefile.some(f => f.statut === 10)
      || data.dossierprimefile.every(f => f.statut === 15 || f.statut === 10)
    : false;

  if (displayButton) {
    const handleClick = () => ending(data.id_dp_operation);

    const loading = pending ? pending.endingLoading : false;
    const inProgress = data.statut ? data.statut.code_statut === 0 : true;

    return (
      <div className="tile is-child">
        <div className="content has-text-centered">
          <button
            onClick={handleClick}
            type="button"
            disabled={!inProgress || locked}
            style={{ transition: 'opacity .2s ease' }}
            className={`button is-primary${loading ? ' is-loading' : ''} is-outlined is-medium`}
          >
            {'Terminer'}
          </button>
        </div>
      </div>
    );
  }

  return null;
};

export default connect(
  null,
  { ending: folderEnding },
)(EndButtonComponent);
