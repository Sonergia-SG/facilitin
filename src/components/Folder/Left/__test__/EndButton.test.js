import React from 'react';
import { shallow } from 'enzyme';

import { EndButtonComponent as EndButton } from '../EndButton';

describe('EndButton', () => {
  it('not render EndButton if dossierprimefile is missing', () => {
    const data = {};
    const wrapper = shallow(<EndButton data={data} locked={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render EndButton if one dpfile is rejected', () => {
    const data = {
      dossierprimefile: [
        {
          statut: 10,
        },
        {
          statut: 0,
        },
      ],
    };
    const wrapper = shallow(<EndButton data={data} locked={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render EndButton if all dpfile is validated', () => {
    const data = {
      dossierprimefile: [
        {
          statut: 15,
        },
        {
          statut: 15,
        },
      ],
    };
    const wrapper = shallow(<EndButton data={data} locked={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('not render EndButton if one dpfile is validated', () => {
    const data = {
      dossierprimefile: [
        {
          statut: 15,
        },
        {
          statut: 0,
        },
      ],
    };
    const wrapper = shallow(<EndButton data={data} locked={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render EndButton enabled if operation status has code "0"', () => {
    const data = {
      statut: {
        code_statut: 0,
      },
      dossierprimefile: [
        {
          statut: 10,
        },
        {
          statut: 0,
        },
      ],
    };
    const wrapper = shallow(<EndButton data={data} locked={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render EndButton disabled if operation status has code "15"', () => {
    const data = {
      statut: {
        code_statut: 15,
      },
      dossierprimefile: [
        {
          statut: 10,
        },
        {
          statut: 0,
        },
      ],
    };
    const wrapper = shallow(<EndButton data={data} locked={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render EndButton disabled if operation status missing "code_statut"', () => {
    const data = {
      statut: {},
      dossierprimefile: [
        {
          statut: 10,
        },
        {
          statut: 0,
        },
      ],
    };
    const wrapper = shallow(<EndButton data={data} locked={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render EndButton loading for pending endingLoading "true"', () => {
    const data = {
      dossierprimefile: [
        {
          statut: 10,
        },
        {
          statut: 0,
        },
      ],
    };
    const wrapper = shallow(
      <EndButton
        data={data}
        pending={{
          endingLoading: true,
        }}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render EndButton not loading for pending endingLoading "false"', () => {
    const data = {
      dossierprimefile: [
        {
          statut: 10,
        },
        {
          statut: 0,
        },
      ],
    };
    const wrapper = shallow(
      <EndButton
        data={data}
        locked={false}
        pending={{
          endingLoading: false,
        }}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render EndButton locked', () => {
    const data = {
      dossierprimefile: [
        {
          statut: 15,
        },
        {
          statut: 15,
        },
      ],
    };
    const wrapper = shallow(<EndButton data={data} locked />);

    expect(wrapper).toMatchSnapshot();
  });
});
