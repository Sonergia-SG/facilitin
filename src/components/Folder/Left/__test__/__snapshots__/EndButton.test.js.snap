// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndButton not render EndButton if dossierprimefile is missing 1`] = `""`;

exports[`EndButton not render EndButton if one dpfile is validated 1`] = `""`;

exports[`EndButton render EndButton disabled if operation status has code "15" 1`] = `
<div
  className="tile is-child"
>
  <div
    className="content has-text-centered"
  >
    <button
      className="button is-primary is-outlined is-medium"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "transition": "opacity .2s ease",
        }
      }
      type="button"
    >
      Terminer
    </button>
  </div>
</div>
`;

exports[`EndButton render EndButton disabled if operation status missing "code_statut" 1`] = `
<div
  className="tile is-child"
>
  <div
    className="content has-text-centered"
  >
    <button
      className="button is-primary is-outlined is-medium"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "transition": "opacity .2s ease",
        }
      }
      type="button"
    >
      Terminer
    </button>
  </div>
</div>
`;

exports[`EndButton render EndButton enabled if operation status has code "0" 1`] = `
<div
  className="tile is-child"
>
  <div
    className="content has-text-centered"
  >
    <button
      className="button is-primary is-outlined is-medium"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "transition": "opacity .2s ease",
        }
      }
      type="button"
    >
      Terminer
    </button>
  </div>
</div>
`;

exports[`EndButton render EndButton if all dpfile is validated 1`] = `
<div
  className="tile is-child"
>
  <div
    className="content has-text-centered"
  >
    <button
      className="button is-primary is-outlined is-medium"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "transition": "opacity .2s ease",
        }
      }
      type="button"
    >
      Terminer
    </button>
  </div>
</div>
`;

exports[`EndButton render EndButton if one dpfile is rejected 1`] = `
<div
  className="tile is-child"
>
  <div
    className="content has-text-centered"
  >
    <button
      className="button is-primary is-outlined is-medium"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "transition": "opacity .2s ease",
        }
      }
      type="button"
    >
      Terminer
    </button>
  </div>
</div>
`;

exports[`EndButton render EndButton loading for pending endingLoading "true" 1`] = `
<div
  className="tile is-child"
>
  <div
    className="content has-text-centered"
  >
    <button
      className="button is-primary is-loading is-outlined is-medium"
      onClick={[Function]}
      style={
        Object {
          "transition": "opacity .2s ease",
        }
      }
      type="button"
    >
      Terminer
    </button>
  </div>
</div>
`;

exports[`EndButton render EndButton locked 1`] = `
<div
  className="tile is-child"
>
  <div
    className="content has-text-centered"
  >
    <button
      className="button is-primary is-outlined is-medium"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "transition": "opacity .2s ease",
        }
      }
      type="button"
    >
      Terminer
    </button>
  </div>
</div>
`;

exports[`EndButton render EndButton not loading for pending endingLoading "false" 1`] = `
<div
  className="tile is-child"
>
  <div
    className="content has-text-centered"
  >
    <button
      className="button is-primary is-outlined is-medium"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "transition": "opacity .2s ease",
        }
      }
      type="button"
    >
      Terminer
    </button>
  </div>
</div>
`;
